TextInputKit/TextInputKit/Code/SpecializedTextInput/CardNumber/BankCardNumberTextInputFormatter.swift

```diff
@@ -20,10 +20,88 @@ final class BankCardNumberTextInputFormatter : TextInputFormatter {
         replacing replacementString: String,
         at editedRange: Range<String.Index>) -> TextInputValidationResult {
 
-        // TODO: Implement.
-        return .accepted
+        let originalString = originalString.unicodeScalars
```

Comment:
But how should I name them?
Initially, I implemented all the formatting using `String`. Then I studied the `String` implementation a little bit and read some materials about Swift `String` performance.
After exploring Swift `String` performance issues my first impulse was to rename all the occurences of `string` in variables, methods and method arguments to `unicodeScalar.`, but it look horrible.